### ISSUE TITLE
Don't use failure

### ISSUE THREAD

**Issue by billop** [dovahcrow/tldextract-rs#3]
`std::error::Error` should be used instead. This can be done manually with some boilerplate or with a library such as snafu or thiserror.

**Comment by dovahcrow** [comment]
Yes you are right. I'm still waiting for the https://github.com/rust-lang/rust/issues/53487 so that I can use `anyhow`. Error without stacktrace is quite painful IMO.
